dcpy/connectors/edm/recipes.py

```diff
@@ -35,26 +37,56 @@ class VersionStrategy(str, Enum):
     bump_latest_release = "bump_latest_release"
 
 
-class Dataset(BaseModel, use_enum_values=True):
+class DatasetType(str, Enum):
+    pg_dump = "pg_dump"
+    csv = "csv"
+
+
+_file_extensions_by_type = {"pg_dump": "sql", "csv": "csv"}
+
+
+class DataPreprocessor(BaseModel, use_enum_values=True, extra="forbid"):
```

Comment:
It's kinda crazy that forbid isn't the default?? Did you find a good reason for that in their docs?

Reply:
I was very surprised by that, as forbidding by default would never lead to silent errors. The alternative does. 

It also seems like something you'd want to address early in the docs, instead of nestling in a [config](https://docs.pydantic.dev/latest/api/config/) section, where it's not super discoverable.